napari/components/add_layers_mixin.py

```diff
@@ -155,91 +187,76 @@ def add_image(
         elif data is None:
             data = io.magic_imread(path)
 
+        # doing this here for IDE/console autocompletion in add_image function.
+        kwargs = {
+            'rgb': rgb,
+            'is_pyramid': is_pyramid,
+            'colormap': colormap,
+            'contrast_limits': contrast_limits,
+            'gamma': gamma,
+            'interpolation': interpolation,
+            'rendering': rendering,
+            'iso_threshold': iso_threshold,
+            'attenuation': attenuation,
+            'name': name,
+            'metadata': metadata,
+            'scale': scale,
+            'translate': translate,
+            'opacity': opacity,
+            'blending': blending,
+            'visible': visible,
+        }
+
+        # these arguments are *already* iterables in the single-channel case.
+        iterable_kwargs = {'scale', 'translate', 'contrast_limits', 'metadata'}
+
         if channel_axis is None:
-            if colormap is None:
-                colormap = 'gray'
-            if blending is None:
-                blending = 'translucent'
-            layer = layers.Image(
-                data,
-                rgb=rgb,
-                is_pyramid=is_pyramid,
-                colormap=colormap,
-                contrast_limits=contrast_limits,
-                gamma=gamma,
-                interpolation=interpolation,
-                rendering=rendering,
-                iso_threshold=iso_threshold,
-                attenuation=attenuation,
-                name=name,
-                metadata=metadata,
-                scale=scale,
-                translate=translate,
-                opacity=opacity,
-                blending=blending,
-                visible=visible,
-            )
-            self.add_layer(layer)
-            return layer
+            kwargs['colormap'] = kwargs['colormap'] or 'gray'
+            kwargs['blending'] = kwargs['blending'] or 'translucent'
+            # Helpful message if someone tries to add mulit-channel kwargs,
+            # but forget the channel_axis arg
+            for k, v in kwargs.items():
+                if k not in iterable_kwargs and is_sequence(v):
+                    raise TypeError(
+                        f"Received sequence for argument '{k}', "
+                        "did you mean to specify a 'channel_axis'? "
+                    )
+
+            return self.add_layer(layers.Image(data, **kwargs))
         else:
-            if is_pyramid:
-                n_channels = data[0].shape[channel_axis]
-            else:
-                n_channels = data.shape[channel_axis]
-
-            name = ensure_iterable(name)
-
-            if blending is None:
-                blending = 'additive'
-
-            if colormap is None:
-                if n_channels < 3:
-                    colormap = colormaps.MAGENTA_GREEN
+            n_channels = (data[0] if is_pyramid else data).shape[channel_axis]
```

Comment:
Won't this fail if we pass in an implicit pyramid? ie pass in a list of arrays and `is_pyramid=None`?

Reply:
lemme see if I can write a test for that